Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ReviewRequestController.java

```diff
@@ -21,12 +21,28 @@
            
     @RequestMapping(value = "/auth/request/add", method = RequestMethod.POST )
     public @ResponseBody ReviewRequest addNewReviewRequest( @ModelAttribute(value="reviewRequest") ReviewRequest reviewRequest,Principal principal) {
-  
         reviewRequest.setStatus(ReviewStatus.OPENED);
         reviewRequest.setRequester(principal.getName());
         
         requestsMgr.addReviewRequest(reviewRequest);
         
         return reviewRequest;
     }
+    
+    @RequestMapping(value = "/auth/request/resolve", method = RequestMethod.POST )
+    public @ResponseBody ReviewRequest resolveRequest(@ModelAttribute(value="reviewRequest") ReviewRequest reviewRequest,Principal principal) {
+        reviewRequest.setStatus(ReviewStatus.RESOLVED);
+        reviewRequest.setResolver(principal.getName());
```

Comment:
Comment will come in the POST body only. It will be pre populated.

Reply:
right. in that case though, why don't you make it more generic and move the status to client site, so this could be used for any kind of status change?